src/OpenGL/Extensions/Silk.NET.OpenGL.Extensions.ImGui/Texture.cs

```diff
@@ -23,55 +23,60 @@ class Texture : IDisposable
         public readonly uint Width, Height;
         public readonly uint MipmapLevels;
         public readonly SizedInternalFormat InternalFormat;
-        public unsafe Texture(GL gl, string name, int width, int height, IntPtr data, bool generateMipmaps = false, bool srgb = false)
+        public unsafe Texture(GL gl, int width, int height, IntPtr data, bool generateMipmaps = false, bool srgb = false)
         {
             _gl = gl;
             MaxAniso ??= gl.GetFloat(MaxTextureMaxAnisotropy);
-            Name = name;
             Width = (uint) width;
             Height = (uint) height;
             InternalFormat = srgb ? Srgb8Alpha8 : SizedInternalFormat.Rgba8;
             MipmapLevels = (uint) (generateMipmaps == false ? 1 : (int)Math.Floor(Math.Log(Math.Max(Width, Height), 2)));
 
-            _gl.CreateTexture(TextureTarget.Texture2D, Name, out GlTexture);
-            _gl.TextureStorage2D(GlTexture, MipmapLevels, InternalFormat, Width, Height);
+            GlTexture = _gl.GenTexture();
+            _gl.BindTexture(GLEnum.Texture2D, GlTexture);
 
-            _gl.TextureSubImage2D(GlTexture, 0, 0, 0, Width, Height, PixelFormat.Bgra, PixelType.UnsignedByte, (void*) data);
+            _gl.TexStorage2D(GLEnum.Texture2D, MipmapLevels, InternalFormat, Width, Height);
+            _gl.TexSubImage2D(GLEnum.Texture2D, 0, 0, 0, Width, Height, PixelFormat.Bgra, PixelType.UnsignedByte, (void*) data);
 
             if (generateMipmaps) _gl.GenerateTextureMipmap(GlTexture);
 
             SetWrap(TextureCoordinate.S, TextureWrapMode.Repeat);
             SetWrap(TextureCoordinate.T, TextureWrapMode.Repeat);
 
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMaxLevel, MipmapLevels - 1);
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureMaxLevel, MipmapLevels - 1);
         }
 
         public void SetMinFilter(TextureMinFilter filter)
         {
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMinFilter, (int)filter);
+            _gl.BindTexture(GLEnum.Texture2D, GlTexture);
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureMinFilter, (int)filter);
         }
 
         public void SetMagFilter(TextureMagFilter filter)
         {
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMagFilter, (int)filter);
+            _gl.BindTexture(GLEnum.Texture2D, GlTexture);
```

Comment:
Can we add a `Bind` method instead so we can chain all these bindings together? Otherwise it'd probably be pretty inefficient when setting lots of properties in one go...

Reply:
Done